### PR TITLE
Added STS auth support to Java client

### PR DESCRIPTION
**Pull request by melanahammel**

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -49,6 +49,8 @@ dependencies {
     compile "com.amazonaws:aws-java-sdk-core:${AWS_SDK_VERSION}"
```
Comment (fieldju) [inline gradle/dependencies.gradle]: Are these other AWS SDK dependencies still used?

File: src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java [hunk 2]
```diff
@@ -240,17 +239,16 @@ protected void getAndSetToken(final String iamPrincipalArn, final Region region)
      * @return Base64 and encrypted token
      */
     protected String getEncryptedAuthData(final String iamPrincipalArn, Region region) {
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java]: This method shouldn't be used any more?

File: src/integration/java/com/nike/cerberus/client/auth/aws/StsAuthCerberusClientTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests StsCerberusCredentialsProvider class
+ */
+public class StsAuthCerberusClientTest {
+
+    private static String region;
```
Comment (fieldju) [inline src/integration/java/com/nike/cerberus/client/auth/aws/StsAuthCerberusClientTest.java]: Why are this static?

File: src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provider for allowing users to authenticate with Cerberus with the STS auth endpoint.
+ */
+
+public class StsCerberusCredentialsProvider extends BaseAwsCredentialsProvider {
+
+    protected static String regionName;
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java]: Why is this static?

File: src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    protected Map<String, String> getSignedHeaders(){
+
+        final String url = "https://sts." + regionName + ".amazonaws.com";
+
+        URI endpoint = null;
+
+        try {
+            endpoint = new URI(url);
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java]: Never e.printStackTrace in production code, this should be log.error("Some message", e)

File: src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java [hunk 6]
```diff
@@ -80,6 +78,8 @@
 
     public static final MediaType DEFAULT_MEDIA_TYPE = MediaType.parse("application/json; charset=utf-8");
 
+    public static final String IAM_ROLE_ARN_FORMAT = "arn:aws:iam::%s:role/%s";
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java]: What is this used for, this seems like logic that was used for non-sts authentication?

File: src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provider for allowing users to authenticate with Cerberus with the STS auth endpoint.
+ */
+
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java]: Extraneous whitespace

**Comment by fieldju** [comment]
There is unused logic in BaseAwsCredentials.
There are unused dependencies in dependencies.gradle (we should only be needing core and sts aws libs)
It seems like EnvironmentUtils should also be deleted.
The build is failing because Findbugs is upset that there are some fields marked static that shouldn't be.

File: src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java (hunk 4 above)
Comment (melanahammel) [reply src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java]: just made it not static, this also seem to fix the failing build

File: src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java (hunk 2 above)
Comment (melanahammel) [reply src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java]: removing this method

File: gradle/dependencies.gradle (hunk 1 above)
Comment (melanahammel) [reply gradle/dependencies.gradle]: testing removing these dependencies

File: src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java (hunk 6 above)
Comment (melanahammel) [reply src/main/java/com/nike/cerberus/client/auth/aws/BaseAwsCredentialsProvider.java]: removed

File: src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java (hunk 7 above)
Comment (melanahammel) [reply src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java]: fixed

File: src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java (hunk 5 above)
Comment (melanahammel) [reply src/main/java/com/nike/cerberus/client/auth/aws/StsCerberusCredentialsProvider.java]: fixed

File: src/integration/java/com/nike/cerberus/client/auth/aws/StsAuthCerberusClientTest.java (hunk 3 above)
Comment (melanahammel) [reply src/integration/java/com/nike/cerberus/client/auth/aws/StsAuthCerberusClientTest.java]: fixed

File: gradle/dependencies.gradle (hunk 1 above)
Comment (melanahammel) [reply gradle/dependencies.gradle]: removed

**Comment by fieldju** [comment]
The tests in `StsCerberusCredentialsProviderTest` are failing because it assumes credentials are available at test time. `credentialsProvider.getSignedHeaders()` needs to be mocked. You can reproduce locally by removing `~/.aws/credentials` or unsetting the env vars.

Also I saw 2 tests in that class that are duplicates, please use a data provider and remove the duplication see: https://github.com/TNG/junit-dataprovider

**Comment by fieldju** [comment]
Oops, didn't mean to press the close PR button.

**Comment by fieldju** [comment]
LGTM, thanks for addressing feedback.

**Comment by melanahammel** [comment]
Added STS auth support to Java client
